frame/assets/src/lib.rs

```diff
@@ -1422,38 +1484,37 @@ impl<T: Config> Pallet<T> {
 				ensure!(&need_admin == &details.admin, Error::<T>::NoPermission);
 			}
 
-			if dest == source || amount.is_zero() {
-				return Ok(())
-			}
+			if dest != source && !amount.is_zero() {
```

Comment:
Need to condition this across most of the function since I moved the deposit_event into here.